### PR TITLE
Model info sanity

### PR DESCRIPTION
**Pull request by halfak**
Some improvements to the way that ModelInfo is handled.  

* Fixes key pattern in format_str for ModelInfo
* Impossible threshold optimizations no longer blow up.
* Adds ModelInfoLookupError and tests.

### CODE REVIEW

File: revscoring/scoring/model_info.py [hunk 1]
```diff
@@ -110,14 +116,15 @@ def format(self, paths=None, formatting="str", **kwargs):
 
     def format_str(self, path_tree, **kwargs):
         formatted = "Model Information:\n"
-        for key in path_tree or self.normalize_fields(path_tree):
-            if hasattr(self[key], "format_str"):
+        for key in self.normalize_fields(path_tree):
+            key_val = try_key(key, self)
```
Comment (halfak) [inline revscoring/scoring/model_info.py]: Note to reviewer, "try_key()" is where the magic happens.

Comment (adamwight) [inline revscoring/scoring/model_info.py]: We no longer test that key_val has .format_str

File: revscoring/scoring/model_info.py [hunk 2]
```diff
@@ -146,11 +153,14 @@ def normalize_fields(self, path_tree):
...
         return d[key]
-    except KeyError as e:
-        if key in ("true", "false"):
-            return d[key == 'true']
-        else:
-            try:
-                return d[int(key)]
-            except ValueError:
-                raise e
+    except KeyError:
```
Comment (adamwight) [inline revscoring/scoring/model_info.py]: __contains__ could call try_key internally, to ensure consistent logic.

File: revscoring/scoring/statistics/classification/tests/test_scaled_threshold_statistics.py [hunk 3]
```diff
@@ -35,6 +35,7 @@ def test_sts():
     print(skewed_sts.format_str({}, threshold_ndigits=3))
     print(balanced_sts.format_str({}, threshold_ndigits=3))
     print(skewed_sts.format_str({'maximum recall @ precision >= 0.9': {}}))
+    print(skewed_sts.format_str({'maximum recall @ precision >= 1.1': {}}))
```
Comment (adamwight) [inline revscoring/scoring/statistics/classification/tests/test_scaled_threshold_statistics.py]: O_o

**Pull request merged**